src/main/kotlin/com/derongan/minecraft/deeperworld/world/Point.kt

```diff
@@ -0,0 +1,18 @@
+package com.derongan.minecraft.deeperworld.world
+
+import kotlin.math.sqrt
+
+/**
+ * Represents a single X/Y/Z cube in a minecraft world.
+ */
+data class CubePoint(val x: Int, val y: Int, val z: Int) {
```

Comment:
Maybe just use the Location or Vector class from spigot

Reply:
Not sure what this offers over just using Int. Tried it but had to use some scuffy ways to calculate all the functions below.